packages/backend/api/src/cocktail.ts

```diff
@@ -7,37 +7,62 @@ import { db } from '@app/backend-shared';
 import type { UpdateData } from '@app/types';
 import type { ActionTable, Flavour } from '@app/types';
 
+import loginIdUser from './middlewares/login-id-user';
 import multerConfig from './middlewares/multer-config';
 import validateCocktailAdd from './middlewares/validate-cocktail-add';
 
-// import loginIdUser from './middlewares/login-id-user';
-
 const cocktailRouter = express.Router();
 
 interface RequestWithUser extends Request {
   userId?: number;
 }
 
-// cocktailRouter.post('/add', loginIdUser, validateCocktailAdd, async (req: RequestWithUser, res: Response) => {
+interface initialData {
+  totalQuantity: number;
+  totalkcal: number;
+  totalComplexity: number;
+  totalDuration: number;
+  quantity: number[];
+  kcal: number[];
+  complexity: number[];
+  duration: number[];
+  flavors: Flavour[];
+  counts: { [key in Flavour]: number };
+  maxCount: number;
+  maxItems: Flavour[];
+}
+
+interface allIngredient {
+  quantity: number;
+  flavour: Flavour;
+  kcal: number;
+}
+
 cocktailRouter.post(
   '/add',
+  loginIdUser,
   validateCocktailAdd,
   async (req: RequestWithUser, res: Response) => {
     const { name, glass, ingredients, alcohol, topping } = req.body;
-    const userId = 1;
-    /* const userId = req.userId; */
-
-    /*  if (userId === undefined) {
-    return res.json({ result: 'not connected' });
-  } */
-    const probability = [
-      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1,
-      1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 4, 4, 5,
-    ];
+    const userId = req.userId;
 
-    const howMuchMoreIngredients = Math.floor(
-      Math.random() * probability.length,
-    );
+    if (userId === undefined) {
+      return res.json({ ok: false, message: 'not connected' });
+    }
+    const random = Math.random();
+    let howMuchMoreIngredients = 0;
+
+    if (random <= 0.4) {
+      howMuchMoreIngredients = 0;
+    } else if (random > 0.4 && random <= 0.6) {
+      howMuchMoreIngredients = 1;
+    } else if (random > 0.6 && random <= 0.8) {
+      howMuchMoreIngredients = 2;
+    } else if (random > 0.8 && random <= 0.9) {
+      howMuchMoreIngredients = 3;
+    } else {
+      howMuchMoreIngredients = 4;
+    }
```

Comment:
Si tu commences par les conditions où la valeur est la plus élevée tu peux simplifier
```suggestion
    let howMuchMoreIngredients = 0;

    if (random > 0.9) { // 10%
      howMuchMoreIngredients = 4;
    } else if (random > 0.8) { // 10%
      howMuchMoreIngredients = 3;
    } else if (random > 0.6) { // 20%
      howMuchMoreIngredients = 2;
    } else if (random > 0.4) { // 20%
      howMuchMoreIngredients = 1;
    } // pas besoin de else vu que par défaut c'est 0 / 40%
```